File: ip-messaging/rest/roles/update-role/update-role.4.x.rb
require 'twilio-ruby'

# Initialize the client
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
ip_messaging_client = Twilio::REST::IpMessagingClient.new account_sid,
                                                          auth_token

# Update a role
options = { Permission: 'leaveChannel' }
role = ip_messaging_client
       .services.get('ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
       .roles.get('RLXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
       .update(options)
puts role

File: ip-messaging/rest/roles/update-role/update-role.4.x.cs
// Download the twilio-csharp library from twilio.com/docs/csharp/install
using System;
using Twilio;

class Example {
    static void Main (string[] args) {
        // Find your Account Sid and Auth Token at twilio.com/user/account
        string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        string authToken = "your_auth_token";
        string serviceSid = "ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        string roleSid = "RLXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        string friendlyName = "friendlyName";

        // Update the role
        List<string> permissions = new List<string>();
        permissions.Add("sendMessage");
        permissions.Add("leaveChannel");

        var client = new TwilioIpMessagingClient(accountSid, authToken);
        Role role = client.UpdateRole(serviceSid, roleSid, friendlyName, permissions);
        Console.WriteLine(role);
    }
}
